library/pkcs7.c

```diff
@@ -87,40 +87,6 @@ static int pkcs7_get_version( unsigned char **p, unsigned char *end, int *ver )
     return( ret );
 }
 
-/**
- * ContentInfo ::= SEQUENCE {
- *      contentType ContentType,
- *      content
- *              [0] EXPLICIT ANY DEFINED BY contentType OPTIONAL }
- **/
-static int pkcs7_get_content_info_type( unsigned char **p, unsigned char *end,
```

Comment:
Instead of getting rid of this function completely, let's rename it `pkcs7_get_content_type`, and Make it only parse the `contentType` field, not the entire ContentInfo Sequence.

That way this function's behaviour will be consistent with all the other `pkcs7_get_*` functions, and we don't end up duplicating as much code.

I.e. remove the first `mbedtls_asn1_get_tag` call, and don't reset the pointer on an error condition.